pageserver/src/tenant/timeline.rs

```diff
@@ -3671,21 +3679,12 @@ impl Timeline {
         target_file_size: u64,
         ctx: &RequestContext,
     ) -> Result<(), CompactionError> {
-        let this = self.clone();
-        let ctx_inner = ctx.clone();
-        let layer_removal_cs_inner = layer_removal_cs.clone();
-        let span = tracing::info_span!("blocking");
         let CompactLevel0Phase1Result {
             new_layers,
             deltas_to_compact,
-        } = tokio::task::spawn_blocking(move || {
-            let _g = span.entered();
-            this.compact_level0_phase1(layer_removal_cs_inner, target_file_size, &ctx_inner)
-        })
-        .await
-        .context("compact_level0_phase1 spawn_blocking")
-        .map_err(CompactionError::Other)
-        .and_then(|res| res)?;
+        } = self
+            .compact_level0_phase1(layer_removal_cs.clone(), target_file_size, ctx)
```

Comment:
It still looks quite CPU heavy to me. So I would keep it in `spawn_blocking` and use `layers.blocking_write` to access layer map

Reply:
We've only been using `spawn_blocking` for 2 weeks or so.
It's more important to get this merged than the potential perf improvement that we got there).

I'll create a follow-up issue to get this back.

Reply:
#4492